### PR TITLE
Substrate to 9251791

### PR DESCRIPTION
**Pull request by dmitrylavrenov**

### CODE REVIEW

File: client/rpc/src/eth/filter.rs [hunk 1]
```diff
@@ -118,19 +119,19 @@ where
...
 		self.create_filter(FilterType::Block)
 	}
 
-	fn new_pending_transaction_filter(&self) -> Result<U256> {
+	fn new_pending_transaction_filter(&self) -> RpcResult<U256> {
 		Err(internal_err("Method not available."))
 	}
 
-	fn filter_changes(&self, index: Index) -> BoxFuture<Result<FilterChanges>> {
+	async fn filter_changes(&self, index: Index) -> RpcResult<FilterChanges> {
```
Comment (MOZGIII) [inline client/rpc/src/eth/filter.rs:134]: We should be able to simplify this fn

File: client/rpc/src/eth_pubsub.rs [hunk 2]
```diff
@@ -250,250 +216,221 @@ where
...
+							// When the node is not under a major syncing (i.e. from genesis), react
+							// normally to import notifications.
+							//
+							// Only send new notifications down the pipe when the syncing status changed.
+							use std::sync::atomic::{AtomicBool, Ordering};
+							let last_syncing_status = AtomicBool::new(Arc::clone(&network).is_major_syncing());
+							let stream = client.clone().import_notification_stream().filter_map(|_| async {
+								let syncing_status = network.is_major_syncing();
+								if syncing_status
+ 									!= last_syncing_status.swap(syncing_status, Ordering::SeqCst)
```
Comment (MOZGIII) [inline client/rpc/src/eth_pubsub.rs:418]: Why is this here again?

Comment (dmitrylavrenov) [reply client/rpc/src/eth_pubsub.rs:418]: Because we merged the same approach in the previous locked.

File: client/rpc/src/eth/filter.rs (hunk 1 above)
Comment (dmitrylavrenov) [reply client/rpc/src/eth/filter.rs:134]: The same, don't think that we should to do it in this PR.

It's better to have a good review later and decide about possible improvements.

File: client/rpc/src/eth_pubsub.rs (hunk 2 above)
Comment (MOZGIII) [reply client/rpc/src/eth_pubsub.rs:418]: Didn't we alter it to get rid of atomics?

**Review by MOZGIII — APPROVED** [review]
Not sure if things are correct here, but we have to move forward with the merge at this time

File: client/rpc/src/eth_pubsub.rs (hunk 2 above)
Comment (dmitrylavrenov) [reply client/rpc/src/eth_pubsub.rs:418]: Let's see final jsonrpsee integration from frontier team or later will propose our implementation.

**Comment by dmitrylavrenov** [comment]
> Not sure if things are correct here, but we have to move forward with the merge at this time

What do you mean _things_are_correct_ ? Everything is the same as it was before. Ok, will merge.

**Comment by MOZGIII** [comment]
I mean that before we skipped the review almost entirely - so before I didn't have any certainty of the correctness of the code. It is mostly trivial changes, except for a few places where the changes are not trivial. So, this time, I've identified those places but didn't go into them yet. Thus - not sure if things are correct, *still*. Precisely as it was with the last PR.
